orttraining/orttraining/eager/opgen/opgen/atenops.py

```diff
@@ -46,57 +46,78 @@ def __init__(self, dY, X):
 ops = {}
 type_promotion_ops = []
 
-for binary_op, onnx_op in {
-    "add": Add("self", Mul("alpha", "other")),
-    "sub": Sub("self", Mul("alpha", "other")),
-    "mul": Mul("self", "other"),
-    "div": Div("self", "other"),
-}.items():
-    for dtype in ["Tensor", "Scalar"]:
-        for variant in ["", "_"]:
-            name = f"aten::{binary_op}{variant}.{dtype}"
-            if name not in ops:
-                ops[f"aten::{binary_op}{variant}.{dtype}"] = deepcopy(onnx_op)
-                type_promotion_ops.append(f"aten::{binary_op}{variant}.{dtype}")
-
-for unary_op in [
+# the following op list is for ops that have a .out version. Often this is the only op needing to be implemented
+# and the regular and inplace(_) version derive from the .out.
```

Comment:
one potential issue is if pytorch guys change the implementation then we might break, as we are not aware of their update until we sync to latest. but that probably the cost we have to pay